pkg/apply/apply.go

```diff
@@ -103,8 +104,11 @@ func DeleteOwnedObject(ctx context.Context, client k8sclient.Client, obj *uns.Un
 	existing.SetGroupVersionKind(gvk)
 	err := client.Get(ctx, types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, existing)
 	if err != nil {
-		if !apierrors.IsNotFound(err) {
-			return errors.Wrapf(err, "could not retrieve existing %s", objDesc)
+		// Fail only if the error is not one of the followings:
+		// - not found (nothing to do since it's already gone)
+		// - the type does not exist (if the resource type does not exist then there is no resource at all)
+		if !apierrors.IsNotFound(err) && !apimeta.IsNoMatchError(err) {
```

Comment:
aint this might mask real stuff of missing schemes ?
maybe safer to limit it just to the specific obj kind / name that you need ?

iiuc lets say you delete a NAD,
we do have the NAD in scheme, if the NAD doesnt exists it will result in IsNotFound
not in IsNoMatchError
what do i miss ? (assumption her that scheme equal to having CRD)

Reply:
we are not talking about scheme, we are talking about what you get from `kubectl api-resources` if nad is not there then for sure there is no instances of it.

Reply:
maybe we want to limit it just for nads ?
or we dont need to worry that it will mask anything as long as we have tests that catch leftovers that should be removed, as you done here

Reply:
we don't need to be nad specific, if the type is not there instance of it are not going to exist, so we good.